### PR TITLE
python310Packages.atomman: 1.4.4 -> 1.4.5

### PR DESCRIPTION
**Pull request by tjni**
In addition, fix up some tests to simplify future maintenance of the build and add a few more comments to explain what I investigated.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/atomman/default.nix [hunk 1]
```diff
@@ -56,13 +65,18 @@ buildPythonPackage rec {
...
-    pytest tests -k "not test_rootdir and not test_version \
-      and not test_atomic_mass and not imageflags \
-      and not test_build_unit and not test_set_and_get_in_units \
-      and not test_set_literal and not test_scalar_model " \
-      --ignore tests/plot/test_interpolate.py \
-      --ignore tests/tools/test_vect_angle.py
+    runHook preCheck
+
+    # pytestCheckHook doesn't work because it imports atomman from the current
+    # directory instead of from where `pip` installs it, and it cannot import
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/atomman/default.nix:71]: Did you try rm'ing that directory or prepending PYTHONPATH with the directory from $out? If that works we prefer that route.

File: pkgs/development/python-modules/atomman/default.nix [hunk 2]
```diff
@@ -21,19 +22,27 @@
...
     repo = "atomman";
     rev = "v${version}";
-    hash = "sha256-iLAB0KMtrTCyGpx+81QfHDPVDhq8OA6CDL/ipVRpyo0=";
+    hash = "sha256-wXz/uHjXKHVKJu/HoFF2mADSBLp6UGF9ivOp2ZOz/Ys=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "fix-tests.patch";
+      url = "https://github.com/usnistgov/atomman/pull/9.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/atomman/default.nix:41]: Fetching entire unmerged PRs is forbidden due to reproducibility. Please fetch each commit.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/atomman/default.nix (hunk 2 above)
Comment (tjni) [reply pkgs/development/python-modules/atomman/default.nix]: Good catch. Fixed.

File: pkgs/development/python-modules/atomman/default.nix (hunk 1 above)
Comment (tjni) [reply pkgs/development/python-modules/atomman/default.nix]: I got it to work by removing the directory. Thank you for the suggestion and, as always, your hard work.

**Pull request merged**